Salt and hash a password with PBKDF2 compliation error

The code in section [Salt and hash a password with PBKDF2](https://rust-lang-nursery.github.io/rust-cookbook/cryptography/encryption.html#salt-and-hash-a-password-with-pbkdf2) results in compilation error.
Here is the errot

```
error[E0308]: mismatched types
  --> src/main.rs:21:9
   |
21 |         N_ITER,
   |         ^^^^^^ expected struct `std::num::NonZeroU32`, found u32
   |
   = note: expected type `std::num::NonZeroU32`
              found type `u32`

error[E0308]: mismatched types
  --> src/main.rs:31:9
   |
31 |         N_ITER,
   |         ^^^^^^ expected struct `std::num::NonZeroU32`, found u32
   |
   = note: expected type `std::num::NonZeroU32`
              found type `u32`

error[E0308]: mismatched types
  --> src/main.rs:39:9
   |
39 |         N_ITER,
   |         ^^^^^^ expected struct `std::num::NonZeroU32`, found u32
   |
   = note: expected type `std::num::NonZeroU32`
              found type `u32`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
error: Could not compile `pbkdf2`.

To learn more, run the command again with --verbose.
```

@AndyGauge it didn't work for me

```
error[E0308]: mismatched types
  --> src/main.rs:56:30
   |
56 |   const N_ITER: NonZeroU32 = 100_000u32;
   |                              ^^^^^^^^^^ expected struct `std::num::NonZeroU32`, found u32
   |
   = note: expected type `std::num::NonZeroU32`
              found type `u32`
```

And if you do this `const N_ITER: NonZeroU32 = NonZeroU32::new(100_000u32).unwrap();`, you'll get another error

```
error[E0015]: calls in constants are limited to tuple structs and tuple variants
  --> src/main.rs:56:30
   |
56 |   const N_ITER: NonZeroU32 = NonZeroU32::new(100_000u32).unwrap();
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: a limited form of compile-time function evaluation is available on a nightly compiler via `const fn`
```

My solution is to use `let n_iter = NonZeroU32::new(100_000u32).unwrap();` and replace `N_ITER` with `n_iter`. Not really sure if that's the correct way to solve the problem tho.

--- a/src/cryptography/encryption/pbkdf2.md
+++ b/src/cryptography/encryption/pbkdf2.md
@@ -17,10 +17,11 @@ use data_encoding::HEXUPPER;
 use ring::error::Unspecified;
 use ring::rand::SecureRandom;
 use ring::{digest, pbkdf2, rand};
+use std::num::NonZeroU32;
 
 fn main() -> Result<(), Unspecified> {
     const CREDENTIAL_LEN: usize = digest::SHA512_OUTPUT_LEN;
-    const N_ITER: u32 = 100_000;
+    let n_iter = NonZeroU32::new(100_000).unwrap();
     let rng = rand::SystemRandom::new();
 
     let mut salt = [0u8; CREDENTIAL_LEN];
@@ -29,8 +30,8 @@ fn main() -> Result<(), Unspecified> {
     let password = "Guess Me If You Can!";
     let mut pbkdf2_hash = [0u8; CREDENTIAL_LEN];
     pbkdf2::derive(
-        &digest::SHA512,
-        N_ITER,
+        pbkdf2::PBKDF2_HMAC_SHA512,
+        n_iter,
         &salt,
         password.as_bytes(),
         &mut pbkdf2_hash,
@@ -39,16 +40,16 @@ fn main() -> Result<(), Unspecified> {
     println!("PBKDF2 hash: {}", HEXUPPER.encode(&pbkdf2_hash));
 
     let should_succeed = pbkdf2::verify(
-        &digest::SHA512,
-		N_ITER,
+        pbkdf2::PBKDF2_HMAC_SHA512,
+        n_iter,
         &salt,
         password.as_bytes(),
         &pbkdf2_hash,
     );
     let wrong_password = "Definitely not the correct password";
     let should_fail = pbkdf2::verify(
-        &digest::SHA512,
-        N_ITER,
+        pbkdf2::PBKDF2_HMAC_SHA512,
+        n_iter,
         &salt,
         wrong_password.as_bytes(),
         &pbkdf2_hash,

